gui-agent/vmside.c

```diff
@@ -964,53 +951,50 @@ static void handle_targets_list(Ghandles * g, Atom Qprop, unsigned char *data,
                     (int) atoms[i], XGetAtomName(g->display,
                         atoms[i]));
     }
-    XConvertSelection(g->display, Clp,
-            have_utf8 ? g->utf8_string_atom : XA_STRING, Qprop,
-            g->stub_win, CurrentTime);
+    XConvertSelection(g->display, g->clipboard,
+            have_utf8 ? g->utf8_string_atom : XA_STRING, g->qprop,
+            g->stub_win, time);
 }
 
+_Thread_local Time the_time;
```

Comment:
gui-agent is a single-thread application, can it be in `Ghandles` instead? In any case, declare global stuff at the beginning of the file.

Reply:
With this change, you don't need extra parameter anymore (functions do receive `g` reference already).